c++/src/kj/compat/http.h

```diff
@@ -1273,6 +1287,12 @@ kj::String generateExtensionRequest(const ArrayPtr<CompressionParameters>& exten
 
 kj::Maybe<CompressionParameters> tryParseExtensionOffers(StringPtr offers);
 
+kj::Maybe<CompressionParameters> tryParseAllExtensionOffers(StringPtr offers,
+    CompressionParameters manualConfig);
+
+kj::Maybe<CompressionParameters> compareClientAndServerConfigs(CompressionParameters requestConfig,
```

Comment:
I should add some tests for these new functions, will do that now.

Reply:
https://github.com/capnproto/capnproto/pull/1505/commits/245d063b095a6c799764b5305cc9f1ca9a979c5d

Test found a subtle bug so I'm happy about that.